### PR TITLE
Do not read block hash of out of index

### LINKED ISSUE #208: BlockChain raises KeyNotFoundException when index is not in it
If there is no value at index when `BlockChain[index]` is called, [ArgumentOutofRangeException](https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception) should occur. 
But it raises ` KeyNotFoundException`, because `Store.IndexBlockHash` returns `default(T)` instead of `null`.

image (4)

### PR DESCRIPTION
**Pull request by gurrpi**
This PR fixes #208. An `ArgumentOutOfRangeException` will be thrown instead of `KeyNotFoundException` when `Blockchain<T>[int]` calls the index of a block that does not exist locally.

### CODE REVIEW

**Comment by longfin** [comment]
@gurrpi Thanks for patch! could you add a mention about this fix on `CHANGES.md`?

**Comment by gurrpi** [comment]
> @gurrpi Thanks for patch! could you add a mention about this fix on `CHANGES.md`?

Sure, I submitted another commits that updates `changelog` and also added test for  `RecentIndex(string @namespace)` — Newly added method by this PR — to suppress Codecov warning.

File: Libplanet/Store/IStore.cs [hunk 1]
```diff
@@ -16,6 +16,8 @@ public interface IStore
 
         long CountIndex(string @namespace);
 
+        long RecentIndex(string @namespace);
```
Comment (dahlia) [inline Libplanet/Store/IStore.cs]: This method doesn't seem used anywhere.  Is this necessary?

Comment (gurrpi) [reply Libplanet/Store/IStore.cs]: Well, of course not. It was a mistake.
I will going to delete the code associated with the method.

**Comment by gurrpi** [comment]
I applied the review comment. Please take a look. @dahlia

**Comment by longfin** [comment]
@gurrpi Thank you for taking the time to work on it. unfortunately, #214 was resolved by another PR (#317) and this PR was closed automatically. 

We still appreciate your work and next contributions.

**Pull request closed without merging**